phpseclib/Crypt/Random.php

```diff
@@ -260,6 +262,32 @@ function crypt_random_string($length)
     }
 }
 
+if (!function_exists('phpseclib_safe_serialize')) {
+    /**
+     * Safely serialize variables
+     *
+     * If a class has a private __sleep() method it'll give a fatal error on PHP 5.2 and earlier.
+     * PHP 5.3 will emit a warning.
+     */
+    function phpseclib_safe_serialize($arr)
+    {
+        if (is_object($arr)) {
+            return '';
+        }
+        if (!is_array($arr)) {
+            return serialize($arr);
+        }
+        $safearr = array();
+        foreach (array_keys($arr) as $key) {
+            if (is_object($arr[$key])) {
+                continue;
+            }
+            $safearr[$key] = is_array($arr[$key]) ? phpseclib_safe_serialize($arr[$key]) : $arr[$key];
```

Comment:
`$GLOBALS` seems to be recursive. So that it seems that some sort recursion prevention would be needed. E.g., I tried this function with `echo phpseclib_safe_serialize($GLOBALS);` on http://sandbox.onlinephpfunctions.com/ and:
* For PHP 4 - 5.2 it doesn't return anything
* For PHP 5.3 - 7.0 it shows:

  ```
  <b>Fatal error</b>:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in <b>[...][...]</b> on line <b>14</b><br />
  ```

This might be dependent on php.ini configuration.

A simple solution of changing line 282 to `if (is_object($arr[$key]) || ($arr[$key] === $GLOBALS)) {` seems to work on higher PHP versions, but not on lower ones. So that maybe a recursion depth limitation could be used as a universal solution? (Like, e.g., in https://secure.php.net/manual/en/function.var-dump.php#76072 or https://secure.php.net/manual/en/function.print-r.php#78851)